### PR TITLE
[WIP] Removing Python 2 related things from the entire codebase

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

#17943 

#### Brief description of what is fixed or changed
1. Removed `integer_types` from compatibility and other files. 
2. Removed `string_types` and `class_types` from compatibility and other files.
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by namannimmo10** [comment]
Hi @asmeurer, please review this before I add more commits?

File: ..b.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+void fun1()
+{
+int a;
+}
```
Comment (asmeurer) [inline ..b.h]: These files should not be here.

Comment (namannimmo10) [reply ..b.h]: I have removed them in the next commit.

**Comment by asmeurer** [comment]
This looks good so far. For most of these the tests would fail if you made a mistake.

In the future, avoid doing `git add *` or whatever it is you did that added those unrelated files.

**Comment by namannimmo10** [comment]
> This looks good so far. For most of these the tests would fail if you made a mistake.
> 
> In the future, avoid doing `git add *` or whatever it is you did that added those unrelated files.

Alright, I'll work on the next task.

**Comment by namannimmo10** [comment]
Wouldn't it be better if merging is done after each task cause if I mess up by any chance, precious work will be destroyed

File: sympy/core/numbers.py [hunk 2]
```diff
@@ -2425,7 +2425,7 @@ def __rfloordiv__(self, other):
         return Integer(Integer(other).p // self.p)
 
 # Add sympify converters
-for i_type in integer_types:
+for i_type in (int, ):
```
Comment (ShubhamKJha) [inline sympy/core/numbers.py]: Probably just
```py
converer[int] = Integer
```

File: sympy/interactive/printing.py [hunk 3]
```diff
@@ -259,7 +258,7 @@ def _result_display(self, arg):
         from sympy.tensor.array import NDimArray
 
         printable_types = [Basic, MatrixBase, float, tuple, list, set,
-                frozenset, dict, Vector, Dyadic, NDimArray] + list(integer_types)
+                frozenset, dict, Vector, Dyadic, NDimArray] + list((int, ))
```
Comment (ShubhamKJha) [inline sympy/interactive/printing.py]:
```suggestion
                frozenset, dict, Vector, Dyadic, NDimArray, int]
```

File: sympy/core/numbers.py (hunk 2 above)
Comment (namannimmo10) [reply sympy/core/numbers.py]: Sure, I'll update. Thanks.

**Comment by namannimmo10** [comment]
> Wouldn't it be better if merging is done after each task cause if I mess up by any chance, precious work will be destroyed

What do you think about this one @ShubhamKJha

**Comment by ShubhamKJha** [comment]
Looks good. Verify no occurrences of `integer_types` remain.
If you don't force-push, or ruin your commit history otherwise, it's highly unlikely you will lose your work. We can go for a merge when all the `_types` variables are removed.

**Comment by asmeurer** [comment]
The git commits keeps your work intact. If you break something in a commit you can always look at the previous commit to see how it was before.

**Comment by namannimmo10** [comment]
OK. Anyway, it's ready for review :) 💪

**Comment by namannimmo10** [comment]
>  We can go for a merge when all the `_types` variables are removed.

Done.

**Comment by namannimmo10** [comment]
@sylee957  still some more tasks are there. Please don't close this.

**Comment by sylee957** [comment]
I’d apologize for jumping in merging
I didn’t see any problems and I thought that you were done for some part.

I don’t think that this should be reverted if it doesn’t introduce any problems. You can open up a new if you just want to add more things.

**Comment by namannimmo10** [comment]
Alright, I'll open up a new one.

**Pull request merged**